x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/activity_log_date_range_picker/index.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useDispatch } from 'react-redux';
+import React, { memo, useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import moment from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiDatePicker, EuiDatePickerRange } from '@elastic/eui';
+
+import * as i18 from '../../../translations';
+import { useEndpointSelector } from '../../../hooks';
+import { getActivityLogDataPaging } from '../../../../store/selectors';
+
+const DatePickerWrapper = styled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+  background: white;
+`;
+const StickyFlexItem = styled(EuiFlexItem)`
+  position: sticky;
+  top: ${(props) => props.theme.eui.euiSizeM};
+  z-index: 1;
+`;
+
+export const DateRangePicker = memo(() => {
+  const dispatch = useDispatch();
+  const { page, pageSize, startDate, endDate } = useEndpointSelector(getActivityLogDataPaging);
+
+  const isInvalidDateRange = useMemo(
+    () => (startDate && endDate ? moment(startDate) > moment(endDate) : false),
+    [startDate, endDate]
+  );
+
+  const onClear = useCallback(
+    ({ clearStart = false, clearEnd = false }: { clearStart?: boolean; clearEnd?: boolean }) => {
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: clearStart ? undefined : startDate,
+          endDate: clearEnd ? undefined : endDate,
+        },
+      });
+    },
+    [dispatch, endDate, startDate, page, pageSize]
+  );
+
+  const onChangeStartDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
+        return;
+      }
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: date ? date?.toISOString() : undefined,
+          endDate: endDate ? endDate : undefined,
+        },
+      });
+    },
+    [dispatch, isInvalidDateRange, endDate, page, pageSize]
+  );
+
+  const onChangeEndDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
```

Comment:
Just in case: Did you ensure that this `isInvalidDateRange` is properly updated when checked in this if? I'm wondering if it has the old value because it depends on `endDate` and `startDate` variables you are going to update below.
Maybe I'm wrong, but I think you are comparing the old value of `isInvalidDateRange`

Reply:
Good catch. 👍  I'll use the new value.

Reply:
done fe043611f64d025e4decb55acae48998328c8c6d